systems/trajectory_optimization/dircon/dircon_mode.h

```diff
@@ -0,0 +1,178 @@
+      std::vector<int> constraint_indices, double scale);
+
+  /// See SetKinVelocityScale(evaluator_index, constraint_index, scale). 
+  /// Sets multiple indices to use the same scale. All constraint_indices within
+  /// all evaluator_indices will use the given scale. This is designed for cases
+  /// where multiple KinematicEvaluators share similar structure, and one wishes
+  /// to give all the same scale factor.
+  void SetKinVelocityScale(std::vector<int> evaluator_indices,
+      std::vector<int> constraint_indices, double scale);
+
+  /// See SetKinAccelerationScale(evaluator_index, constraint_index, scale). 
+  /// Sets multiple indices to use the same scale. All constraint_indices within
+  /// all evaluator_indices will use the given scale. This is designed for cases
+  /// where multiple KinematicEvaluators share similar structure, and one wishes
+  /// to give all the same scale factor.
+  void SetKinAccelerationScale(std::vector<int> evaluator_indices,
+      std::vector<int> constraint_indices, double scale);
+
+  /// Gets the map of non-unitary scales for the dynamics constraints
+  const std::unordered_map<int, double>& GetDynamicsScale() const {
+    return dynamics_scale_;
+  };
+
+  /// Gets the map of non-unitary scales for the impact constraints
+  const std::unordered_map<int, double>& GetImpactScale() const {
+    return impact_scale_;
+  };
+
+  /// Gets the map of non-unitary scales for the kinematic position constraints
+  /// Note, unlike SetKinPositionScale, indices in the map refer to the complete
+  /// KinematicEvaluatorSet, not a specific KinematicEvaluator.
+  const std::unordered_map<int, double>& GetKinPositionScale() const {
+    return kin_position_scale_;
+  };
+
+  /// Gets the map of non-unitary scales for the kinematic velocity constraints
+  /// Note, unlike GetKinVelocityScale, indices in the map refer to the complete
+  /// KinematicEvaluatorSet, not a specific KinematicEvaluator.
+  const std::unordered_map<int, double>& GetKinVelocityScale() const {
+    return kin_velocity_scale_;
+  };
+
+
+  /// Gets the map of non-unitary scales for the kinematic acceleration 
+  /// constraints. Note, unlike GetKinAccelerationScale, indices in the map 
+  /// refer to the complete KinematicEvaluatorSet, not a specific
+  /// KinematicEvaluator.
+  const std::unordered_map<int, double>& GetKinAccelerationScale() const {
+    return kin_acceleration_scale_;
+  };
+
+  /// Get the cost on force regularization. Defaults to 1e-4
+  double get_force_regularization() const { return force_regularization_;};
+
+ private:
+  void SetKinScale(std::unordered_map<int, double>* scale_map,
+      int evaluator_index, int constraint_index, double scale);
+ 
+  const multibody::KinematicEvaluatorSet<T>& evaluators_;
+  int num_knotpoints_;
+  double max_T_;
+  double min_T_;
+  std::set<int> relative_constraints_;
+  std::unordered_map<int, DirconKinConstraintType> reduced_constraints_;
```

Comment:
Is this variable used to keep track of the constraint type of the kinematics constraints? If so, the name could be a bit confusing